src/draw/lv_draw_rect.h

```diff
@@ -154,6 +178,14 @@ void lv_draw_box_shadow_dsc_init(lv_draw_box_shadow_dsc_t * dsc);
  */
 lv_draw_box_shadow_dsc_t * lv_draw_task_get_box_shadow_dsc(lv_draw_task_t * task);
 
+/**
+ * Draw a box shadow
+ * @param layer         pointer to a layer
+ * @param dsc           pointer to an initialized draw descriptor variable
+ * @param coords        the coordinates of the rectangle
+ */
+void lv_draw_box_shadow(lv_layer_t * layer, const lv_draw_box_shadow_dsc_t * dsc, const lv_area_t * coords);
```

Comment:
same here with the deviation from the API.

Reply:
> same here with the deviation from the API.

Some drawing functions have a coords parameter, while others do not. Perhaps it would be better to standardize them.
![image](https://github.com/user-attachments/assets/9ec9ea68-fd13-409a-9a3f-0d226b55b7a4)
![image](https://github.com/user-attachments/assets/71c55feb-11e0-474a-9d69-2363b079441c)
![image](https://github.com/user-attachments/assets/f7f52f77-a423-45e8-8b45-11c7033156d3)

Reply:
That's what I am thinking should be done. Either add the points, areas, coords or other location based information in the draw functions call arguments or locate them in the descriptor structures. One or the other should be done and it should be done for all types of drawing types. It's always a good thing to keep the API consistent.